src/utils/wallet.js

```diff
@@ -48,7 +57,7 @@ export function generateWordsToValidate(numWordsToGenerate: number, maxWords: nu
   return chosenWords;
 }
 
-export async function getSaltedPin(pin: string, dispatch: Function): Promise<string> {
+export async function getSaltedPin(pin: string, dispatch: Dispatch): Promise<string> {
```

Comment:
This function feels weird because of getting that redux `Dispatch`. Consider changing the parameters to `deviceUniqueId, pin` and doing the `DeviceInfo` checks in the call site.